src/org/traccar/protocol/LdplProtocolDecoder.java

```diff
@@ -0,0 +1,107 @@
+                .expression("([0|1]),")              // sos_alert
+                .expression("([0|1]),")              // body_tamper
+                .expression("([0|1])([0|1]),")       // ac_status + ign_status
+                .expression("([0|1|2]),")            // output1_status
+                .number("(d{1,3}),")                 // adc1
+                .number("(d{1,3}),")                 // adc2
+                .expression("([0-9A-Z]{3}),")        // software_version
+                .expression("([L|R]),")              // message_type
+                .expression("([0-9A-Z]{4})#")        // crc
+                .compile();
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        Parser parser = new Parser(PATTERN, (String) msg);
+        if (!parser.matches()) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.set(Position.KEY_TYPE, parser.nextInt());
+
+        if (!identify(parser.next(), channel, remoteAddress)) {
+            return null;
+        }
+        position.setDeviceId(getDeviceId());
+
+        DateBuilder dateBuilder = new DateBuilder()
+        .setDateReverse(parser.nextInt(), parser.nextInt(), parser.nextInt())
+        .setTime(parser.nextInt(), parser.nextInt(), parser.nextInt());
+        position.setTime(dateBuilder.getDate());
+
+        if ("A".equals(parser.next())) {
+            position.setValid(true);
+        } else {
+            position.setValid(false);
+        }
+        
+        position.setLatitude(parser.nextCoordinate(CoordinateFormat.DEG_MIN_MIN_HEM));
+        position.setLongitude(parser.nextCoordinate(CoordinateFormat.DEG_MIN_MIN_HEM));
+        
+        position.setSpeed(parser.nextDouble());
+        position.setCourse(parser.nextDouble());
+        
+        position.set(Position.KEY_SATELLITES, parser.nextInt());
+        position.set(Position.KEY_GSM, parser.nextInt());
+        String vehicleStatus = parser.next();
+        position.set(Position.KEY_POWER, parser.nextInt());
+        position.set(Position.KEY_BATTERY, parser.nextDouble());
+        position.set(Position.KEY_ALARM, parser.nextInt());
+        Integer bodyTamper = parser.nextInt();
+        Integer acStatus = parser.nextInt();
+        position.set(Position.KEY_IGNITION, parser.nextInt());
+        position.set(Position.KEY_OUTPUT, parser.nextInt());
+        position.set(Position.PREFIX_ADC + 1, parser.nextInt());
+        position.set(Position.PREFIX_ADC + 2, parser.nextInt());
+        position.set(Position.KEY_VERSION, parser.next());
+        
+        if ("R".equals(parser.next())) {
+            position.set(Position.KEY_ARCHIVE, true);
+        } else {
+            position.set(Position.KEY_ARCHIVE, false);
+        }
+        
+        String checksum = parser.next();
```

Comment:
Yes, I decided to not check CRC because all protocols use TCP and UDP transport, both of which include integrity check.